include/deal.II/fe/fe_bernstein.h

```diff
@@ -0,0 +1,266 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2000 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef __deal2__fe_bernstein_h
+#define __deal2__fe_bernstein_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/tensor_product_polynomials.h>
+#include <deal.II/fe/fe_q_base.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup fe */
+/*@{*/
+
+/**
+ * Implementation of a scalar Bernstein finite element @p that we call FE_Bernstein
+ * in analogy with FE_Q that yields the
+ * finite element space of continuous, piecewise Bernstein polynomials of degree @p p in
+ * each coordinate direction. This class is realized using tensor product
+ * polynomials of Bernstein basis polynomials.
+ *
+ *
+ * The standard constructor of this class takes the degree @p p of this finite
+ * element.
+ *
+ * For more information about the <tt>spacedim</tt> template parameter
+ * check the documentation of FiniteElement or the one of
+ * Triangulation.
+ *
+ * <h3>Implementation</h3>
+ *
+ * The constructor creates a TensorProductPolynomials object that includes the
+ * tensor product of @p Bernstein polynomials of degree @p p. This
+ * @p TensorProductPolynomials object provides all values and derivatives of
+ * the shape functions.
+ *
+ * <h3>Numbering of the degrees of freedom (DoFs)</h3>
```

Comment:
The "Numbering of the degrees of freedom (DoFs)" comes from the description of the <code>FE_Q</code> element. There is no need to repeat it here in full. If necessary, reference to the documentation fo this class.